packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet.tsx

```diff
@@ -252,27 +278,23 @@ export function DashboardWalletCreateDialog(props: WrappedDialogProps<object>) {
                 children: (
                     <div>
                         <TextField
-                            helperText={
-                                checkInputLengthExceed(name)
-                                    ? t('input_length_exceed_prompt', {
-                                          name: t('wallet_name').toLowerCase(),
-                                          length: WALLET_OR_PERSONA_NAME_MAX_LEN,
-                                      })
-                                    : undefined
-                            }
+                            helperText={errors.name?.type === zod.ZodIssueCode.too_big && errors?.name?.message}
                             required
                             autoFocus
+                            name="name"
                             label={t('wallet_name')}
-                            value={name}
-                            onChange={(e) => setName(e.target.value)}
+                            inputRef={register}
+                            defaultValue={name.current}
+                            onChange={(e) => (name.current = e.target.value)}
                             variant="outlined"
                         />
                         <TextField
                             type="password"
                             required
                             label={t('private_key')}
-                            value={privKey}
-                            onChange={(e) => setPrivKey(e.target.value)}
+                            key="private_password"
```

Comment:
Why a `key` field here but absence in others?

Reply:
There is a problem here, if u don't add `key` , enter the value in mnemnoic input and switch to private key tab. This value still exists.

The detail reason: https://react-hook-form.com/faqs#Whyisdefaultvaluenotchangingcorrectlywithternaryoperator

Reply:
I see, react reuse the DOM node.